merlion/models/automl/autoprophet.py

```diff
@@ -7,10 +7,19 @@
 """
 Automatic (multi)-seasonality detection for Facebook's Prophet.
 """
-from typing import Union
+import copy
+import logging
+from typing import Any, Iterator, Optional, Tuple, Union
 
+import numpy as np
+
+from merlion.evaluate.forecast import ForecastMetric
+from merlion.models.anomaly.forecast_based.base import ForecastingDetectorBase
 from merlion.models.automl.seasonality import PeriodicityStrategy, SeasonalityConfig, SeasonalityLayer
```

Comment:
Should we extract the class for periodicity detection and move the implementation out from autosarima_utils in the next step? Then, we can include more various periodicity detection strategy.

Reply:
Good idea. Let me move periodicity detection to the `SeasonalityLayer`.